pubsub/consumer.go

```diff
@@ -46,6 +47,9 @@ type Consumer[Request any, Response any] struct {
 	redisStream string
 	redisGroup  string
 	cfg         *ConsumerConfig
+	// terminating indicates whether interrupt was received, in which case
+	// consumer should clean up for graceful shutdown.
+	terminating atomic.Bool
```

Comment:
now this var is not needed, store(true) only happens after the checking thread is done, so a true will never be read.

Reply:
Done.